File: src/Application Specific/Classes/Entity Collections/Checklist/SubTask/SubTaskValidator.vb
Namespace Entity

Namespace SubTasks

Public Class SubTaskValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oSubTask As SubTask)

        'make sure that contact object is valid
        If oSubTask.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oSubTask.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
                End If

                If oSubTask.Description.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Description Missing")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Checklist/SubTask/SubTaskValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace SubTasks
    {
        // make sure that contact object is valid
        public class SubTaskValidator : BaseValidator
        {
            public void Validate(SubTask oSubTask)
            {
                // make sure that contact object is valid
                if (oSubTask.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oSubTask.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oSubTask.Description.Trim().Length == 0)
                {
                    AddCriticalMessage("Description Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}